ion-schema/src/main/kotlin/com/amazon/ionschema/model/constraints/AnnotationsConstraint.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazon.ionschema.model.constraints
+
+import com.amazon.ionschema.model.AstType
+import com.amazon.ionschema.model.ConstraintId
+
+data class AnnotationsConstraint(override val type: AstType) : UnaryTypeConstraint<AnnotationsConstraint> {
```

Comment:
Shouldn't this take a list of allowed symbols/annotations instead of `AstType`? 
Maybe have a new class representing `Annotation` with `required`,`closed` or `ordered` properties, which could be used here.

Reply:
I've chosen to use something that aligns with the proposed syntax in https://github.com/amzn/ion-schema/issues/51, assuming that the proposal will be adopted.